### PR TITLE
Make some erros more descriptive and human-friendly

### PR DESCRIPTION
**Pull request by koiuo**
Would appreciate suggestions on exact wording

### CODE REVIEW

**Review by hellmanj — APPROVED** [review]

**Review by amanjeev — APPROVED** [review]

File: acctest/main_test.go [hunk 1]
```diff
@@ -83,7 +83,24 @@ func TestShouldFailIfServerIsNotListening(t *testing.T) {
...
+	}
+	defer srv.GracefulStop()
+
+	// when
+	stdout, stderr, exitcode := runBin(t, stubSrvAddr)
+
+	// then
+	assert.Equal(t, 127, exitcode)
+	assert.Empty(t, stdout)
+	assert.Equal(t, stderr, "rpc error: server doesn't implement gRPC health-checking protocol\n")
```
Comment (ashald) [inline acctest/main_test.go:103]: Should we mention concrete version this utility supports? It's `grpc.health.v1` after all.

File: acctest/main_test.go [hunk 2]
```diff
@@ -173,7 +190,7 @@ func TestShouldFailOnTlsVerificationWithSelfSignedCert(t *testing.T) {
 	// then
 	assert.Equal(t, 127, exitcode)
 	assert.Empty(t, stdout)
-	assert.Contains(t, stderr, "rpc error")
+	assert.Contains(t, stderr, "TLS handshake")
```
Comment (ashald) [inline acctest/main_test.go:193]: Well, if we log `TLS handshake` upon issue with TLS verification it's already good, but it's still hard to guess how exactly the error message looks like from this test.

**Review by olgert — APPROVED** [review]
LGTM

File: acctest/main_test.go (hunk 1 above)
Comment (koiuo) [reply acctest/main_test.go:103]: I wouldn't worry about it, until something else than v1 is released (which, I bet, will likely never happen).  After all, it is just a human-readable message and we can change it anytime.

If someone is about to rely on exact wording of an error message, I'd suggest then to rely on exit code instead. We can introduce new ones if needed later

File: acctest/main_test.go (hunk 2 above)
Comment (koiuo) [reply acctest/main_test.go:193]: On one hand, I didn't want to specify exact message in test, on the other, having test with exact wording really helps to understand, what is the expected behavior.

I'm leaning towards updating all tests to verify exact messages. If no objections to that, will commit shortly.

Comment (ashald) [reply acctest/main_test.go:193]: Well, I'm not sure if the whole message is really required but at least a portion of it that describes the error. Details like path to the cert or its fingerprint are less interesting from the test perspective, IMO. But either way is OK with me.

Comment (koiuo) [reply acctest/main_test.go:193]: just to clarify, we can't distinct from grpc api between connection refused and TLS handshake failed. So the message will be same for both.

Comment (ashald) [reply acctest/main_test.go:193]: Maybe it worth including this remark into the error message so that the end user will be aware of detail?

**Pull request merged**